packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -891,6 +898,80 @@ class FileCollection extends DocumentCollection {
       data: normalizeFile(resp.data)
     }
   }
+
+  /**
+   * async findNotSynchronizedDirectories - Returns the list of directories not synchronized on the given OAuth client (mainly Cozy Desktop clients) — see https://docs.cozy.io/en/cozy-stack/not-synchronized-vfs/#get-datatypedoc-idrelationshipsnot_synchronizing
+   *
+   * @param  {OAuthClient} oauthClient  A JSON representing an OAuth client, with at least a `_type` and `_id` field.
+   * @param  {object} options           Additional options
+   * @param  {number} options.skip      For skip-based pagination, the number of referenced files to skip.
+   * @param  {number} options.limit     For pagination, the number of results to return.
+   * @param  {object} options.cursor    For cursor-based pagination, the index cursor.
+   * @returns {object}                  The JSON API conformant response.
+   */
+  async findNotSynchronizedDirectories(
+    oauthClient,
+    { skip = 0, limit, cursor } = {}
+  ) {
+    const params = {
+      include: 'files',
+      'page[limit]': limit,
+      'page[cursor]': cursor,
+      sort: 'id'
+    }
+    const url = uri`/data/${oauthClient._type}/${oauthClient._id}/relationships/not_synchronizing`
+    const path = querystring.buildURL(url, params)
+    const resp = await this.stackClient.fetchJSON('GET', path)
+    return {
+      data: resp.data.map(f => normalizeFile(f)),
+      included: resp.included ? resp.included.map(f => normalizeFile(f)) : [],
+      next: has(resp, 'links.next'),
+      meta: resp.meta,
+      skip
+    }
+  }
+
+  /**
+   *  Add directory synchronization exclusions to an OAuth client — see https://docs.cozy.io/en/cozy-stack/not-synchronized-vfs/#post-datatypedoc-idrelationshipsnot_synchronizing
+   *
+   *  For example, to exclude directory `/Photos` from `My Computer`'s desktop synchronization:
+   * ```
+   * addNotSynchronizedDirectories({_id: 123, _type: "io.cozy.oauth.clients", clientName: "Cozy Drive (My Computer)", clientKind: "desktop"}, [{_id: 456, _type: "io.cozy.files", name: "Photos", path: "/Photos"}])
+   * ```
+   *
+   * @param  {OAuthClient} oauthClient  A JSON representing the OAuth client
+   * @param  {Array}  directories       An array of JSON documents having a `_type` and `_id` fields and representing directories.
+   * @returns {object}                  The JSON API conformant response.
+   */
+  addNotSynchronizedDirectories(oauthClient, directories) {
+    const refs = directories.map(d => ({ id: d._id, type: d._type }))
+    return this.stackClient.fetchJSON(
+      'POST',
+      uri`/data/${oauthClient._type}/${oauthClient._id}/relationships/not_synchronizing`,
+      { data: refs }
```

Comment:
no need to normalizeFile the response?

Reply:
See https://github.com/cozy/cozy-client/pull/953#discussion_r646556452